### ISSUE TITLE
NSClientV3 Full sync fails when encounters non-sgv entries

### ISSUE THREAD

**Issue by justmara** [nightscout/AndroidAPS#2763]
[Here](https://github.com/nightscout/cgm-remote-monitor/issues/8090) is the discussion started in NS issue.
tl;dr
/entries/history api can return non-sgv records

This [line of code](https://github.com/nightscout/AndroidAPS/blob/dev/core/ns-sdk/src/main/java/info/nightscout/sdk/mapper/SvgMapper.kt#L17) results in null record when there is no sgv field in entry.

And [here](https://github.com/nightscout/AndroidAPS/blob/dev/core/ns-sdk/src/main/java/info/nightscout/sdk/NSAndroidClientImpl.kt#L125) all the nulls are filtered out 🙂

So when Full sync tries to iterate over entries it stops prematurely because it receives lower amount of records than it requested. But these records were filtered-out on NSClient side.

**Comment by MilosKozak** [comment]
resolved in https://github.com/nightscout/AndroidAPS/commit/d379b769d3739b75bb02d07d300f41871cb18c6d
